### DISCUSSION TITLE
Example with predicate in drift file query seems outdated

### DISCUSSION THREAD

**Discussion by fgblomqvist** [simolus3/drift#2275] [Q&A] [answered]
I have a query in a drift file that follows this example:
https://drift.simonbinder.eu/docs/using-sql/drift_files/#dart-components-in-sql
Where the query takes a predicate that it uses in the where-claus.

After upgrading from Moor 4.3 to Drift 2.0 the query no longer takes an `Expression<bool>` but rather a function which has the two tables in my query as args:
```
Expression<bool> Function($AccountsTable accounts, $CurrenciesTable currencies)
```
I guess the example should be updated?

**Comment by simolus3** [comment, marked as answer]
Thanks for the report! I've updated the sources of the documentation in 2c63c1a64e6288e7fea6bf745fff172d63bdd3cc, the change will go live with the next drift release.

This change is related to the "Using column names in Dart" warning currently displayed below that snippet in the documentation. When you have lots of tables (or, in particular, the same table available under two different names), finding the right column to use in Dart can be a bit challenging. With the new generation mode, all tables that are available at the position of the Dart template are passed as parameters which makes the method safer to use.

This was a breaking change from drift 1.x (=moor 4.x) to drift 2.x. You can keep the old behavior with the `scoped_dart_components: false` [builder option](https://drift.simonbinder.eu/docs/advanced-features/builder_options/), but I recommend migrating for the safety benefits.

**Reply by fgblomqvist** [reply]
Makes sense, yeah migrating was not a problem once I realized what was going on. Thanks for updating!
